app/routes/webhooks.shop.redact.tsx

```diff
@@ -0,0 +1,18 @@
+import type { ActionFunctionArgs } from "@remix-run/node";
+import { authenticate } from "~/shopify.server";
+
+export const action = async ({ request }: ActionFunctionArgs) => {
+  const { session } = await authenticate.webhook(request);
+
+  // SHOP_REDACT will be fired up to 48 hours after app is uninstalled
+  // Therefore, for SHOP_REDACT we expect the admin to be undefined
+  if (!session) {
```

Comment:
This is is the scenario where even if we don't have a session, or access to the admin we still want to process the webhook. (We probably won't have a session because deleted it in response to the `APP_UNINSTALLED` event)

Apps may need to clean up other tables in storage, or make calls to delete information from external services.